react/week3/todolist/src/components/App.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import Form from './Form';
+import TodoList from './TodoList';
+import uuid from 'uuid';
+
+class App extends React.Component {
+   constructor() {
+      super();
+      this.state = {
+         isLoading: true,
+         todos: [],
+         id: uuid(),
+         done: true,
+         error: null,
+      };
+      this.handleSubmit = this.handleSubmit.bind(this);
+      this.handleChangeTodo = this.handleChangeTodo.bind(this);
+      this.handleChangeDate = this.handleChangeDate.bind(this);
+
+       this.handleCheckbox = this.handleCheckbox.bind(this);
+       this.removeTodo = this.removeTodo.bind(this);
+       this.handleEdit = this.handleEdit.bind(this);
+   }
+
+   fetchTodos() {
+      fetch(
+         `https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw`
+      )
+         .then((response) => response.json())
+         .then((data) =>
+            this.setState({
+               todos: data,
+               isLoading: false,
+            })
+         )
+         .catch((error) => this.setState({ error, isLoading: false }));
```

Comment:
Since you always want to see `isLoading` to false regardless of whether it resolves or rejects, you could set it in a finally block instead! 
```suggestion
      fetch(
         `https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw`
      )
         .then((response) => response.json())
         .then((data) =>
            this.setState({todos: data})
         )
         .catch((error) => this.setState({ error}))
         .finally(() =>  this.setState({ isLoading: false }));
```

Reply:
Also consider using [async/await](https://javascript.info/async-await) inside a try/catch blocked instead to improve readability